src/controllers/sign_up.js

```diff
@@ -1,3 +1,64 @@
-exports.get = (req,res) =>{
-    res.render('sign_up',{ style:"style", dom:"sign-up", title:"sign up",a:true});
+const hashPassword = require('./hashPassword');
+const { addUser } = require('../model/queries/users');
+const { addPerson } = require('../model/queries/person');
+const { addBusiness } = require('../model/queries/business');
+
+exports.get = (req, res) => {
+  res.render('sign_up', {
+    style: 'style', dom: 'sign_up', vald: 'validation', title: 'sign up', a: true,
+  });
+};
+
+exports.post = (request, response) => {
+  const data = request.body;
+  const { email } = data;
+  const { password } = data;
+  const { type } = data;
+
+  if (type == 0 || 1 && email && password) {
+    hashPassword(password, (err, hash) => {
+      if (err) {
+        response.render('sign_up', { dom: 'sign_up', msg: 'ERROR' });
+      } else {
+        addUser(email, hash, type)
+          .then((res) => {
+            const user_id = res.rows[0].id;
+            if (type == 0) { // type = 0 represents person
+              const { userName } = data;
+              const { fName } = data;
+              const { lName } = data;
+              const { birthDay } = data;
+              const { gender } = data;
+
+              addPerson(user_id, userName, fName, lName, birthDay, gender)
```

Comment:
Too many parameters. They can all be stored in one object and given as one parameter to the function.